File: python/447_回旋镖的数量_medium.py
class Solution:
    def numberOfBoomerangs(self, points: List[List[int]]) -> int:
        ans = 0
        for p in points:
            cnt = defaultdict(int)
            for q in points:
                dist = (p[0]-q[0])*(p[0]-q[0]) + (p[1]-q[1])*(p[1]-q[1])
                cnt[dist] += 1
            for m in cnt.values():
                ans += m*(m-1)
        return ans

File: cpp/447_回旋镖的数量_medium.cpp
class Solution {
public:
    int numberOfBoomerangs(vector<vector<int>>& points) {
        int ans = 0;
        for (auto &p : points) {
            unordered_map<int, int> cnt;
            for (auto &q : points) {
                int dist = (p[0]-q[0])*(p[0]-q[0]) + (p[1]-q[1])*(p[1]-q[1]);
                ++cnt[dist];
            }
            for (auto &[_, m] : cnt)
                ans += m*(m-1);
        }
        return ans;
    }
};